### PR TITLE
Simplified Contributing guides

### PR DESCRIPTION
**Pull request by laughingbiscuit**
As discussed with @seymen and @danistrebel - a set of simplified Contributing steps and separation of pipeline details.

- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,130 +1,36 @@
...
+
+- [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo)
+ the repository and make your contribution - please don't make changes to
+ multiple projects in the same Pull Request!
+- Ensure that your project has a [`pipeline.sh`](./PIPELINES.md) in the root of
+ your project - see [here](./PIPELINES.md) for more
+- Put the [Apache](https://opensource.google/docs/releasing/preparing/#Apache-header)
+ license header in your source files (you might [automate](https://github.com/google/addlicense)
+ this!)
+- Put a new line in the [CODEOWNERS](./CODEOWNERS) file to declare [ownership](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners)
```
Comment (danistrebel) [inline CONTRIBUTING.md:21]: Maybe indicate that this only applies to new solutions and not small fixes etc.

File: PIPELINES.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Pipelines
+
+Each project should have a `pipeline.sh` script that will be run nightly to
+install and test your code and alert if there are regression failures.
+
+## pipeline.sh
+
+The `pipeline.sh` script:
+
+- is run whenever a Pull Request changes that projects
```
Comment (danistrebel) [inline PIPELINES.md:10]: trivial `project`

File: PIPELINES.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ and [In Solidarity](https://github.com/apps/in-solidarity) are triggered
+ automatically and do not need to be included in your pipeline script. We
+ recommend you allow these to run in GitHub as it is simpler than running
+ locally. We don't mind if the Pull Request fails at first due to these checks!
+
+## GitHub Pages
+
+If you would like to generate static HTML as part of your project, you can
+create a `generate-docs.sh` script. This script should generate static HTML in
+ the subdirectory `./generated/docs`, after which it will be deployed to GitHub
```
Comment (danistrebel) [inline PIPELINES.md:45]: after which => from where?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,130 +1,36 @@
...
+- We follow the [Google Open Source Community Guidelines](https://opensource.google/conduct/)
+
+## Quickstart
+
+- [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo)
+ the repository and make your contribution - please don't make changes to
+ multiple projects in the same Pull Request!
+- Ensure that your project has a [`pipeline.sh`](./PIPELINES.md) in the root of
+ your project - see [here](./PIPELINES.md) for more
+- Put the [Apache](https://opensource.google/docs/releasing/preparing/#Apache-header)
```
Comment (danistrebel) [inline CONTRIBUTING.md:18]: trivial `Apache` => `Apache 2.0`

**Review by danistrebel — COMMENTED** [review]
Minor comments.

File: PIPELINES.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+install and test your code and alert if there are regression failures.
+
+## pipeline.sh
+
+The `pipeline.sh` script:
+
+- is run whenever a Pull Request changes that project
+- is run for all projects each night
+- is run on a private continuous integration server after an initial code
+ review, to reduce the risk of exposing secrets
```
Comment (seymen) [inline PIPELINES.md:13]: let's not reveal too much here. How about `can be triggered manually at any time by ??`

**Review by seymen — CHANGES_REQUESTED** [review]

**Review by seymen — APPROVED** [review]

**Pull request merged**